xbmc/utils/URIUtils.cpp

```diff
@@ -517,32 +509,57 @@ bool URIUtils::IsProtocol(const std::string& url, const std::string &type)
   return StringUtils::StartsWithNoCase(url, type + "://");
 }
 
-bool URIUtils::PathStarts(const std::string& url, const char *start)
+bool URIUtils::PathHasParent(const std::string& path, const std::string& parent, bool translate /* = false */)
```

Comment:
@Montellese : Initially I had those as copies (not refs), but some how the "path = CSpecialProtocol::TranslatePath(path)"/"parent = CSpecialProtocol::TranslatePath(parent)" caused crashes. That's why I made it the way it is now. You understand why that doesn't work, one should be able to pass something by ref and at the same return into that same variable the result or is that illegal?